aws-batch-using-nice-dcv.dockerfile

```diff
@@ -45,53 +43,41 @@ RUN wget http://us.download.nvidia.com/tesla/418.87/NVIDIA-Linux-x86_64-418.87.0
-    nice-dcv-2020.0-8428-el7/nice-dcv-gl-2020.0.759-1.el7.i686.rpm \
-    nice-dcv-2020.0-8428-el7/nice-dcv-gltest-2020.0.229-1.el7.x86_64.rpm \
-    nice-dcv-2020.0-8428-el7/nice-dcv-gl-2020.0.759-1.el7.x86_64.rpm \
-    nice-dcv-2020.0-8428-el7/nice-dcv-server-2020.0.8428-1.el7.x86_64.rpm \
-    nice-dcv-2020.0-8428-el7/nice-xdcv-2020.0.296-1.el7.x86_64.rpm
+                              --no-kernel-module-source \
+ && rm -f /tmp/NVIDIA-installer.run \
+ && nvidia-xconfig --preserve-busid 
+RUN rpm --import https://s3-eu-west-1.amazonaws.com/nice-dcv-publish/NICE-GPG-KEY \
+ && mkdir -p /tmp/dcv-inst \
+ && cd /tmp/dcv-inst \
+ && wget -qO- https://d1uj6qtbmh3dt5.cloudfront.net/2020.0/Servers/nice-dcv-2020.0-8428-el7.tgz |tar xfz - --strip-components=1 \
+ && yum -y install \
+    nice-dcv-gl-2020.0.759-1.el7.i686.rpm \
+    nice-dcv-gltest-2020.0.229-1.el7.x86_64.rpm \
+    nice-dcv-gl-2020.0.759-1.el7.x86_64.rpm \
+    nice-dcv-server-2020.0.8428-1.el7.x86_64.rpm \
+    nice-xdcv-2020.0.296-1.el7.x86_64.rpm
+
+# Define the dcvserver.service
+COPY dcvserver.service /usr/lib/systemd/system/dcvserver.service
 
-# Define the dcvserver.service and enable graphical target
-RUN echo "[Unit]" > /usr/lib/systemd/system/dcvserver.service && \
-    echo "Description=NICE DCV server daemon" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "[Service]" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "PermissionsStartOnly=true" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "ExecStartPre=-/sbin/modprobe eveusb" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "ExecStart=/usr/bin/dcvserver -d --service" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "Restart=always" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "BusName=com.nicesoftware.DcvServer" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "User=dcv" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "Group=dcv" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "[Install]" >> /usr/lib/systemd/system/dcvserver.service && \
-    echo "WantedBy=multi-user.target" >> /usr/lib/systemd/system/dcvserver.service && \
-    systemctl set-default graphical.target
+# Define the service to automatically start the dcv sessions for test users
+COPY dcvcreatesessions.service /usr/lib/systemd/system/dcvcreatesessions.service
+
+# Disable audit, configure Xorg, start DCV server
+COPY run_script.sh /usr/local/bin/
+
+# Open required port on firewall, start a DCV session for user
+RUN echo "firewall-cmd --zone=public --permanent --add-port=8443/tcp" >> "/etc/rc.local" \
+ && echo "firewall-cmd --reload" >> "/etc/rc.local" \
+ && echo "/bin/dcv create-session --owner user001 --user user001 user001session" >> /etc/rc.local \
+ && chmod +x "/etc/rc.local"
 
+CMD ["/usr/local/bin/run_script.sh"]

```

Comment:
@afalzonelab That is how Batch is going to know how to run that script. You do not need to specify anything in the job definition. You could specify this again tho. Does not hurt - but in case you change something in the container I'd say it is cleaner if the container specifies what command to execute and not the job definition.